### ISSUE TITLE
Mismatched Cursor on HiDPI display

### ISSUE THREAD

**Issue by dallenbaldwin** [a85/POSTMan-Chrome-Extension#10030]
**Describe the Issue**
On Postman for Linux v8.5.1 Snap, using a 3000x2000 display, the cursor does not match the system cursor and is _very_ small.

**Steps to Reproduce**
1. Download Postman for Linux v8.5.1 via the Snap Store 
2. Select a different system cursor
3. Open Postman
4. Behold the ultra tiny cursor that doesn't match the system cursor

**Screenshots or Videos**
![Screenshot from 2021-06-15 11-48-51](https://user-images.githubusercontent.com/14098845/122099961-d925f580-cdcf-11eb-8c18-52be0a46c25c.png)
![Screenshot from 2021-06-15 11-49-04](https://user-images.githubusercontent.com/14098845/122099962-d9be8c00-cdcf-11eb-95e5-d5c6fc0b1dbd.png)

**Environment Information**

 - Platform Type: Linux 
 - Postman Version: 8.5.1 Snap
 - Operating System: Linux (Zorin OS 15.3 x86_64 + GNOME 3.30.2)

**Additional Context**
Manually downloading and using the Postman binaries doesn't seem to have this issue.

**Comment by abhinavg267** [comment]
Is there any update for this issue?

**Comment by jviotti** [comment]
Thanks a lot for the detailed report. We'll be taking a look at this very soon!

**Comment by itsrafsanjani** [comment]
@jviotti Is there any update on this? I'm having the same problem with Ubuntu 20.04 LTS.

**Comment by johntria** [comment]
The problem remain with me in Ubuntu 20.04 LTS.

**Comment by RaisinTen** [comment]
Which version of postman are y'all using btw?

**Comment by johntria** [comment]
> Which version of postman are y'all using btw?

Postman for Linux

Version 9.15.2

Architecture x64

Platform linux 5.13.0-39-generic

**Comment by RaisinTen** [comment]
Does it happen only in the snap app or are you able to reproduce it when you download it from the website too -  https://www.postman.com/downloads/?

**Comment by RaisinTen** [comment]
Also, does this help?
1. add
    ```sh
    export XCURSOR_PATH=$RUNTIME/usr/share/icons
    ```
    to `~/.profile`
2. terminate postman and any chrome process running in the background
3. relaunch the postman app

Also wondering if this happens in the Debian version of the app that can be installed from the `Ubuntu Software` app.

**Comment by johntria** [comment]
As i can see from the source code it works properly but from Ubuntu Software the problem remain with us.
I added ` export XCURSOR_PATH=$RUNTIME/usr/share/icons` but the issue not resolved.

**Comment by RaisinTen** [comment]
This appears to be the same issue as reported in https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1579268. I don't think we can do much about this other than just waiting for Snap to roll out a fix but I'll wait for a while for the rest of my team to confirm.

**Comment by RaisinTen** [comment]
cc @postmanlabs/desktop-foundation

**Comment by johntria** [comment]
You are right, I uninstalled all packages from snap-store and implemented by PPA.
Everything works as excepted now. 

Thank you for your time !

**Comment by RaisinTen** [comment]
@johntria sure, no worries!
I was also wondering if you could try to set `XCURSOR_PATH` to `/snap/postman/current/usr/share/icons` instead and try to relaunch the snap postman app. The reason why I'm suggesting this is that `RUNTIME` might have been just an empty string (it is in my case). This should make the app attempt to load the installed cursor theme to match the system theme instead of loading the tiny X11 one (see https://github.com/ubuntu/snapcraft-desktop-helpers/pull/97 for reference).

**Comment by deveshXm** [comment]
Problem persists in Ubuntu 22.04 LTS

**Comment by erkinsergey** [comment]
Also in Mint 20.1

**Comment by hholst80** [comment]
This seems to be a universal problem with Electron inside a Flatpak (or snap) container.

1. Open up the flatpak environment.
2. Set the environment variables
3. Run Postman

We do not know what in the internals of Postman that goes wrong, but a developer should be able to attach a debugger to the process or sprinkle the code with some console.log and find out where it is trying to find the resources to run with a larger cursor shape than the default. 

Telegram Desktop packaged in the same environment configuration is indeed able to use a custom XCURSOR_SIZE *if* XCURSOR_THEME=Adwaita. 

So, the question is where is Telegram looking where Electron and Postman is forgetting to look? :)

PS. The non-sandboxed Postman works just fine and handles a custom XCURSOR_SIZE setting. 

https://user-images.githubusercontent.com/6200749/194727294-d87ca627-159c-4626-a940-0f24b26931d6.mp4

**Comment by hholst80** [comment]
I think I've gone as far as I will with this. Election uses Chromium for the UI stuff. Chromium in a flatpak will suffer from the same limitation as Postman and other Electron based apps. 

Unless some kind of platform solution (workaround) is found for Chromium I don't see how any progress be made as Electron is Limited to what Chromium can support.

https://chromium-review.googlesource.com/c/chromium/src/+/2378142

**Comment by googsvg** [comment]
@jviotti @RaisinTen 
any updates for this?

problem still remains for Snap version

OS:
Ubuntu 22.04.1 LTS

Snap:
Postman for Linux
Version 8.12.5
linux 5.15.0-52-generic / x64

**Comment by kerbrek** [comment]
Maybe developers need to enable "content interface plugs" https://forum.snapcraft.io/t/how-to-use-the-system-gtk-theme-via-the-gtk-common-themes-snap/6235

For example, the output of the `snap connections rocketchat-desktop` command for "Rocket.Chat" (the cursor is displayed correctly in the application):
```
Interface                 Plug                                       Slot                             Notes
...
content[gtk-3-themes]     rocketchat-desktop:gtk-3-themes            gtk-common-themes:gtk-3-themes   -
content[icon-themes]      rocketchat-desktop:icon-themes             gtk-common-themes:icon-themes    -
content[sound-themes]     rocketchat-desktop:sound-themes            gtk-common-themes:sound-themes   -
...
```

Firefox had a similar problem recently. Although "content interface plugs" were present, they were not enabled (connected) after installation. Therefore, the cursor was [small](https://askubuntu.com/questions/1430107/tiny-cursor-in-firefox-snap/) and [looked ugly](https://askubuntu.com/questions/1432707/how-can-i-get-back-the-default-cursor-over-links-in-ubuntu-22-04-with-firefox-sn/).

**Comment by paneverka** [comment]
The problem remains for Snap:

OS: Ubuntu 23.10
Snap: Postman for Linux
Version: 10.22.9

**Comment by DanielNoga** [comment]
The problem remains for Snap:

OS: Kubuntu 24.04
Snap: Postman for Linux
Version: 10.24.16

**Comment by angelof-exe** [comment]
The problem remains for Snap:

OS: Zorin OS 17.1 Core
Snap: Postman for Linux
Version: 11.5.0

**Comment by zscao** [comment]
> The problem remains for Snap:
> 
> OS: Zorin OS 17.1 Core Snap: Postman for Linux Version: 11.5.0

Same here, I am on Manjaro.

**Comment by cammarb** [comment]
The problem remains for the snap version.
I'm using Ubuntu 24.04.

**Postman details**

- Version: 11.8.1
- UI version: 11.8.1-ui-240814-0827
- Desktop platform version: 11.8.0
- Architecture: x64
- OS platform: linux 6.8.0-41-generic

**Comment by someperson** [comment]
This is still a major problem for me. As there is no apt repository available anywhere I can find for Ubuntu 24.04, I am unsure how I am supposed to keep this application up to date without jumping through a lot of hoops.

**Comment by tyleris-irish** [comment]
Issue still present on Ubuntu 24.04 snap
- Version: 11.33.0
- UI version: 11.33.0-ui-250216-2333
- Desktop platform version: 11.33.0
- Architecture: x64
- OS platform: linux 6.8.0-53-generic

**Comment by klo2k** [comment]
This work-around fixed it for me on Ubuntu 24.04.3, postman 10.24.26.

```bash
gsettings set org.gnome.desktop.interface cursor-theme 'DMZ-Black'
```

Bit of a sledgehammer approach but ¯\_(ツ)_/¯ .

**Comment by cuonglmptit** [comment]
I recently ran into the same issue on Hyprland with Postman installed through Flatpak. It's likely you are running Postman through XWayland instead of native Wayland, and you don't have a fallback cursor set up for XWayland.

Step 1: Create the icon config file by running mkdir -p ~/.icons/default && nano ~/.icons/default/index.theme and paste:
[Icon Theme]
Inherits=your_cursor_theme_name
Save it and exit.

Step 2: Restart Postman.

If it still does not work, follow these optional steps:

Step 3: Open Flatseal, select Postman, go to the Filesystem section, and under Other files grant read-only access to where your cursor theme is installed (for example, /usr/share/icons:ro or ~/.local/share/icons:ro).

Step 4: In the Environment section of Flatseal, add the variables XCURSOR_THEME=your_cursor_theme_name and XCURSOR_SIZE=24. You can also adjust the cursor size value here to your liking, as long as your cursor theme supports it (for example, 24, 26, or 32).

Step 5: Restart Postman again.
